modules/overriding-default-node-ip-selection-logic.adoc

```diff
@@ -0,0 +1,142 @@
+
+.Example
+
+[source,terminal]
+----
+$ ip route show
+----
++
+Example output:
++
+[source,terminal]
+----
+default via 10.0.0.1 dev eth0 proto dhcp src 10.0.0.10 metric 426
+10.0.0.0/24 dev eth0 proto kernel scope link src 10.0.0.10 metric 48
+192.0.2.0/24 dev eth1 proto kernel scope link src 192.0.2.5 metric 48
+----
+
+By default, the `10.0.0.10` IP address is selected for use by the node.
+
+To override the default IP selection logic, you can create a hint file that includes the `NODEIP_HINT` variable to override the default IP selection logic. Creating a hint file allows users to select a specific node IP address from the interface in the subnet of the IP address specified in the `NODEIP_HINT` variable. For example, users can configure the `NODEIP_HINT` variable to point at an IP of `192.0.2.1` so that the correct subnet, `192.0.2.0/24`, is selected. As a result, the `192.0.2.5` IP address on `eth1` is used for the node. 
+
+Use the following procedure to override the default node IP selection logic. 
+
+.Procedure
+
+. Reveal the available IP routes of the node by entering the following command: 
++
+[source,terminal]
+----
+$ ip route show
+----
++
+Example output:
++
+[source,terminal]
+----
+default via 10.0.0.1 dev eth0 proto dhcp src 10.0.0.10 metric 426
+10.0.0.0/24 dev eth0 proto kernel scope link src 10.0.0.10 metric 48
+192.0.2.0/24 dev eth1 proto kernel scope link src 192.0.2.5 metric 48
+----
+
+. Use the `runtimecfg` command to reveal the interface that can reach the IP routes:
++
+[source,terminal]
+----
+$ runtimecfg 192.0.2.1 
+----
++
+Example output:
++
+[source,terminal]
+----
+192.0.2.1 dev eth1 src 192.0.2.5 uid 1000
+    cache
+----
+
+. Add a hint file to your your `/etc/default/nodeip-configuration` file. The contents of your hint file must match the following example: 
++
+[source,text]
+----
+NODEIP_HINT=192.0.2.5
+----
++
+[IMPORTANT]
+====
+* Do not use the exact IP address of a node as a hint, for example, `192.0.2.1`. Using the exact IP address of a node causes the node using the hint IP address to fail to configure correctly.
+* The IP address in the hint file is only used to determine the correct subnet. It will not receive traffic as a result of appearing in the hint file. 
+====
+
+. Generate the `base-64` encoded content:
++
+[source,terminal]
+----
+$ echo 'NODEIP_HINT=192.0.2.1' | base64
+----
++
+Example output:
++
+[source,terminal]
+----
+Tk9ERUlQX0hJTlQ9MTkyLjAuMCxxxx==
+----
+
+. Activate the hint by creating a machine config manifest for both `master` and `worker` roles before deploying the cluster: 
++
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+ labels:
+   machineconfiguration.openshift.io/role: master
+   name: 99-nodeip-hint-master
+spec:
+ config:
+   ignition:
+     version: 3.2.0
+   storage:
+     files:
+     - contents:
+         source: data:text/plain;charset=utf-8;base64, <encoded_content> <1>
```

Comment:
This suggest we don' t need `base64` encode, we can use plain text

https://github.com/openshift/machine-config-operator/pull/2888#issuecomment-1005912433

> I used EnvironmentFile in my example because it's a lot easier for the user that way. The MachineConfig just needs to say:

```
source: data:,KUBELET_NODEIP_HINT=192.0.2.1
```